aten/src/ATen/native/sparse/cuda/SparseCUDATensor.cu

```diff
@@ -152,4 +193,100 @@ SparseTensor coalesce_sparse_cuda(const SparseTensor& self) {
   return dst;
 }
 
+Tensor sparse_mask_helper_cuda(
+    const SparseTensor& t,
+    const Tensor& mask_indices) {
+  /*
+    This is a helper function which filter values from `t._values()` using the
+    `mask_indices`. This CUDA implementation uses `thrust::lower_bound`
+    operation to find the intersection of the `mask_indices` and the
+    `t._indices()` to then filter the values.
+
+    Inputs:
+      `t`             - coalesced sparse tensor input
+      `mask_indices`  - mask indices tensor
+  */
+  TORCH_CHECK(t.is_sparse(), "t: input is not a sparse tensor");
+  TORCH_CHECK(t.is_coalesced(), "t:  input is uncoalesced");
+  TORCH_CHECK(mask_indices.dim() == t._indices().dim(), "mask_indices: operands have incompatible indices dim; self has dim ",
+      t._indices().dim(), " but mask has dim ", mask_indices.dim());
+  TORCH_CHECK(mask_indices.is_contiguous(), "mask_indices: mask is not contiguous");
+
+  int64_t r_nnz = mask_indices.size(1);
+  auto t_values = t._values().contiguous();
+  auto full_size = t.sizes();
+  auto vsize = t_values.sizes().vec();
+  vsize[0] = r_nnz;
+
+ 
+  if (t.sparse_dim() == 0) {
+    Tensor t_values_expand = t_values;
+    t_values_expand = t_values_expand.expand(vsize).clone(at::MemoryFormat::Contiguous);
+    return t_values_expand;
+  } 
+  Tensor r_values = at::zeros({vsize}, t_values.options());
+  auto t_indices = t._indices().contiguous();
+  auto t_nnz = t._nnz();
+
+  cudaStream_t stream = at::cuda::getCurrentCUDAStream();
+  auto allocator = THCThrustAllocator(globalContext().lazyInitCUDA());
+  auto policy = thrust::cuda::par(allocator).on(stream);
+
+  // Step 1: flatten the sparse indices `t._indices()` tensor into a 1D indices
+  // tensor `t_flatten_indices`.
+  auto t_flatten_indices = at::sparse::flatten_indices(t_indices, full_size).contiguous();
+
+  // Step 2: flatten the sparse indices `mask_indices` tensor into a 1D indices
+  // tensor `mask_flatten_indices`. Note: This could be not sorted if the input
+  // indices in the constructor are not in a coalesced form
+  auto flattened_mask_indices =
+      at::sparse::flatten_indices(mask_indices, full_size);
+
+  Tensor t_indices_pos = at::empty({r_nnz}, mask_indices.options());
+
+  // Step 3: Match the flattened `mask_indices` with the flattened
+  // `t._indices()` using the `thrust::lower_bound`
+  thrust::lower_bound(
+      policy,
+      t_flatten_indices.data_ptr<int64_t>(),
+      t_flatten_indices.data_ptr<int64_t>() + t_nnz,
+      flattened_mask_indices.data_ptr<int64_t>(),
+      flattened_mask_indices.data_ptr<int64_t>() + r_nnz,
+      t_indices_pos.data_ptr<int64_t>());
+
+  // Step 4: Copy the Filtered `t._values()` using the matches at `t_indices_pos`
+  if (r_nnz > 0 && t_values.numel() > 0) {
+    const dim3 block = dim3(
+        std::min(static_cast<int64_t>(cuda::getApplyBlock().x), r_nnz));
+    dim3 grid;
+    int curDevice = -1;
+    cudaGetDevice(&curDevice);
+    TORCH_CHECK(
```

Comment:
Avoid using raw cuda APIs like cudaGetDevice, and getApplyGrid/Block (it will be deprecated), in this case you can use `at::cuda::getCurrentDeviceProperties()->maxGridSize[0],` to get max grid size, and ATenCeilDiv. Since your grid is 1D you don't even need comparing against maxGridSize (it's 2**31).